src/Simple.OData.Client.V4.Adapter/ResponseReader.cs

```diff
@@ -216,7 +216,7 @@ private ODataEntryAnnotations CreateAnnotations(ODataResource odataEntry)
             {
                 try
                 {
-                    id = odataEntry.Id.AbsoluteUri;
+                    id = odataEntry.Id?.AbsoluteUri;
```

Comment:
Why was this needed as part of this change?

Reply:
OData service generates for aggregated entites ids that are nulls (s. http://docs.oasis-open.org/odata/odata-json-format/v4.0/errata03/os/odata-json-format-v4.0-errata03-os-complete.html#_Toc453766634) so this change was necessary to avoid NullReferenceException.

Reply:
Gotcha, so basically this line from the spec?
> If the entity is transient (i.e. cannot be read or updated), the odata.id annotation MUST appear and have the null value.

Reply:
yes, right

Reply:
If you wouldn't mind adding a comment for that spec link. As this project expands to cover more and more of the OData features it becomes harder to figure out exactly why certain design decisions are made.